src/main/java/nuffle/ui/Ui.java

```diff
@@ -0,0 +1,141 @@
+package nuffle.ui;
+
+import nuffle.exception.NuffleException;
+import nuffle.task.Task;
+import java.util.ArrayList;
+
+/**
+ * The Ui class handles the interaction with the user.
+ * It displays messages to the user based on the commands executed.
+ */
+public class Ui {
+
+    /**
+     * Prints a line separator to the console.
+     */
+    private static void printLine() {
+        System.out.println("---------------------------------------------");
+    }
+
+    /**
+     * Displays the welcome message when the application starts.
+     */
+    public static void welcomeMessage() {
+        // Greeting the users
+        printLine();
+        System.out.println("Nuffle > Good day! I'm Nuffle.");
+        System.out.println("Nuffle > What can I do for you today?");
+        printLine();
+    }
+
+    /**
+     * Displays a message indicating that a task has been marked as done.
+     *
+     * @param task The task that has been marked as done.
+     */
+    public static void markTaskMessage(Task task) {
+        printLine();
+        System.out.println("Nice! I have marked this task as done!");
+        System.out.println(" " + task);
+        printLine();
+    }
+
+    /**
+     * Displays an error message if there was an issue with marking a task.
+     */
+    public static void markTaskError() {
+        printLine();
+        System.out.println("Opps! There appears to be an index error!");
+        printLine();
+    }
+
+    /**
+     * Displays a message indicating that a task has been marked as not done.
+     *
+     * @param task The task that has been marked as not done.
+     */
+    public static void unmarkTaskMessage(Task task) {
```

Comment:
perhaps these print statements could be placed under the `TaskList` class, as it is just a print statement and might not require its own method